va/caa_test.go

```diff
@@ -637,6 +637,62 @@ func TestCAAFailure(t *testing.T) {
 	test.AssertEquals(t, prob.Type, probs.CAAProblem)
 }
 
+func TestFilterCAA(t *testing.T) {
+	testCases := []struct {
+		name              string
+		input             []*dns.CAA
+		expectedIssueVals []string
+		expectedWildVals  []string
+		expectedCU        bool
+	}{
+		{
+			name: "happy path",
+			input: []*dns.CAA{
+				{Tag: "issue", Value: "a"},
+				{Tag: "issuewild", Value: "b"},
+			},
+			expectedIssueVals: []string{"a"},
+			expectedWildVals:  []string{"b"},
+		},
```

Comment:
I think a few more tests should be added here, something like this:

In the event that there's a CAA record with an issue, issuewild, and an iodef record that is [marked as critical](https://www.rfc-editor.org/rfc/rfc6844#section-5.1) we should still expect the CAA check to pass due to the `case "iodef": continue` you added.
```
		{
			name: "happy path with a critical iodef",
			input: []*dns.CAA{
				{Tag: "issue", Value: "a"},
				{Tag: "issuewild", Value: "b"},
				{Tag: "iodef", Value: "c", Flag: 128},
			},
			expectedIssueVals: []string{"a"},
			expectedWildVals:  []string{"b"},
		},
```

and a test for an iodef record without the critical flag
```
		{
			name: "happy path with a regular iodef",
			input: []*dns.CAA{
				{Tag: "issue", Value: "a"},
				{Tag: "issuewild", Value: "b"},
				{Tag: "iodef", Value: "c"},
			},
			expectedIssueVals: []string{"a"},
			expectedWildVals:  []string{"b"},
		},
```


One last thing, since the switch statement default checks for critical flag as either 128 or 1, we should include both of those values in a test.

Reply:
> In the event that there's a CAA record with an issue, issuewild, and an iodef record that is [marked as critical](https://www.rfc-editor.org/rfc/rfc6844#section-5.1) we should still expect the CAA check to pass due to the `case "iodef": continue` you added.

Yes, this case is covered by the "recognized criticals" test case below, in which the iodef is marked as critical.

> and a test for an iodef record without the critical flag

Added to the first test case.

> One last thing, since the switch statement default checks for critical flag as either 128 or 1, we should include both of those values in a test.

Added.